### PR TITLE
docs: add more description for max_connection_duration in timeout faqs

### PR DESCRIPTION
**Pull request by agrawroh**
This PR adds some more description around `max_connection_duration` in the Timeout FAQs.

**Commit Message:** add more description for max_connection_duration in timeout faqs.
**Additional Description:** -
**Risk Level:** N/A
**Testing:** N/A
**Docs Changes:** Added some more description around `max_connection_duration` in the Timeout FAQs.
**Release Notes:** N/A
**Platform Specific Features:** N/A

**Signed-off-by:** Rohit Agrawal <rohit.agrawal@databricks.com>

### CODE REVIEW

File: api/envoy/config/core/v3/protocol.proto [hunk 1]
```diff
@@ -149,8 +149,9 @@ message HttpProtocolOptions {
 
   // The maximum duration of a connection. The duration is defined as a period since a connection
   // was established. If not set, there is no max duration. When max_connection_duration is reached
-  // the connection will be closed. Drain sequence will occur prior to closing the connection if
-  // if's applicable. See :ref:`drain_timeout
+  // the connection will be closed. Any inflight requests will still get processed correctly if this
+  // timeout fires during request processing. Drain sequence will occur prior to closing the connection
+  // if it's applicable. See :ref:`drain_timeout
```
Comment (alyssawilk) [inline api/envoy/config/core/v3/protocol.proto:154]: I think it's not quite accurate to say that all in flight requests will be processed.
Maybe rephrase here and below to make it clear that if there's no active streams, the connection will be closed and if there are active streams, the drain sequence will kick off and the connection will be force-closed after the drain period?

**Review by alyssawilk — COMMENTED** [review]
Nice catch - thanks for making sure this one got documented!

File: api/envoy/config/core/v3/protocol.proto (hunk 1 above)
Comment (agrawroh) [reply api/envoy/config/core/v3/protocol.proto]: Thanks, @alyssawilk. I rephrased it as per your suggestions.

File: api/envoy/config/core/v3/protocol.proto [hunk 2]
```diff
@@ -149,10 +149,11 @@ message HttpProtocolOptions {
 
   // The maximum duration of a connection. The duration is defined as a period since a connection
   // was established. If not set, there is no max duration. When max_connection_duration is reached
-  // the connection will be closed. Drain sequence will occur prior to closing the connection if
-  // if's applicable. See :ref:`drain_timeout
+  // the connection will be closed. If there are no active streams, the connection will be closed.
```
[the remark this answers is not in the document]
Comment (agrawroh) [reply api/envoy/config/core/v3/protocol.proto]: Done.

File: docs/root/faq/configuration/timeouts.rst [hunk 3]
```diff
@@ -30,6 +30,21 @@ Connection timeouts apply to the entire HTTP connection and all streams the conn
   connections use the
   :ref:`common_http_protocol_options <envoy_v3_api_field_extensions.upstreams.http.v3.HttpProtocolOptions.common_http_protocol_options>` field in the Cluster's :ref:`extension_protocol_options<envoy_v3_api_field_config.cluster.v3.Cluster.typed_extension_protocol_options>`, keyed by `envoy.extensions.upstreams.http.v3.HttpProtocolOptions`
 
+* The HTTP protocol :ref:`max_connection_duration <envoy_v3_api_field_config.core.v3.HttpProtocolOptions.max_connection_duration>`
+  is defined in a generic message used by both the HTTP connection manager as well as upstream cluster
+  HTTP connections but is currently only implemented for the downstream connections. The maximum
+  connection duration is the time after which a downstream connection will be closed, starting from when
```
[the remark this answers is not in the document]
Comment (agrawroh) [reply docs/root/faq/configuration/timeouts.rst]: Done.

**Review by alyssawilk — APPROVED** [review]
Awesome, thanks!

**Comment by alyssawilk** [comment]
@markdroth latest version LGTY?
